scipy/stats/stats.py

```diff
@@ -7469,6 +7470,214 @@ def brunnermunzel(x, y, alternative="two-sided", distribution="t",
     return BrunnerMunzelResult(wbfn, p)
 
 
+def _confint_lowerbound(n, quantile, confidence):
+    """Computes the lower bound for a one-sided confidence interval
+    for a given
+    - quantile (0<`quantile`<1)
+    - confidence level (0<`confidence`<1)
+    - number of samples `n`.
+
+    Returns the largest index of the sample being a valid lower bound,
+    or `None` if there are not enough samples to derive one.
+
+    Used by the public function confint_quantile().
+
+    .. versionadded:: 1.6.0
+    """
+    # compute all probabilities from the binomiale distribution for the quantile of interest
+    bd = binom(n, quantile)
+    ppm = [np.maximum(1-x, 0.0) for x in np.cumsum([bd.pmf(k) for k in range(n)])]
+
+    # search the index defining the lower-bound
+    if ppm[0] < confidence:
+        return None
+    else:
+        for k in range(n):
+            # search for first index reaching below the desired confidence
+            if ppm[k] < confidence:
+                # lower-bound is the previous index
+                return k-1
+
+
+def confint_quantile(x, quantile, confidence, type='one-sided'):
+    r"""Computes non-parametric confidence intervals for any quantile
+    from a set of i.i.d. samples.
```

Comment:
I had to google `i.i.d. samples`--I suppose I'm not really the target audience though :)